src/migrations/20240313000002-delete_comlog_entry.js

```diff
@@ -0,0 +1,130 @@
+const {
+  prepMigration,
+} = require('../lib/migration');
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, _Sequelize) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      const sessionSig = __filename;
+      await prepMigration(queryInterface, transaction, sessionSig);
+      await queryInterface.sequelize.query(
+        `
+        -- Delete communications log records as desired by a customer.
+        -- Because there's no soft delete column (deletedAt), we're:
+        
+        -- Doing a hard DELETE FROM in the communications log and link tables (currently just CommunicationLogFiles)
+        -- Removing any Files that only exist due to the Communication Log.
+
+        -- The File removal step is intensive because currently there are many different ways
+        -- for a File to be linked and all have to be checked before the File can be removed.
+
+        -- Create the sets of things to be deleted and before counts
+        DROP TABLE IF EXISTS before_set;
+        CREATE TEMP TABLE before_set
+        AS
+        SELECT DISTINCT
+          cl.id clid,
+          clf.id clfid,
+          clf."fileId" fid
+        FROM "CommunicationLogs" cl
+        LEFT JOIN "CommunicationLogFiles" clf
+          ON clf."communicationLogId" = cl.id
+        WHERE cl.id IN (788, 888);
+        
+
+        DROP TABLE IF EXISTS before_count;
+        CREATE TEMP TABLE before_count
+        AS
+        SELECT
+          COUNT(DISTINCT clid) comlog_count,
+          COUNT(DISTINCT clfid) comlogfile_count,
+          COUNT(DISTINCT fid) file_count
+        FROM before_set;
+
+        -- Check to see if any files are only present because of this communication log
+        -- unfortunately there's currently many ways to link a file
+        DROP TABLE IF EXISTS files_to_delete;
+        CREATE TEMP TABLE files_to_delete
+        AS
+        WITH candidate_files AS (SELECT DISTINCT fid FROM before_set),
+        file_links AS (
+        SELECT
```

Comment:
Because this is getting intensive, should we make a helper function that check all tables with a foreign key to the files table base on a supplied list. That way we know they will all always be checked and properly handled?

Reply:
I guess we could do that if we're not going to do the generic refactor.

Reply:
refactor does not look to be happening any time soon